### PR TITLE
fix aria svg color - collapsible component now uses react aria disclosure.

### PR DESCRIPTION
**Pull request by fraincs**
- Fixed color of WAG-Aria SVG color in doc [383]
- Updated collapsible doc component to use React Aria Disclosure [385]

### CODE REVIEW

File: apps/docs/components/collapsible/Collapsible.tsx [hunk 1]
```diff
@@ -1,54 +1,31 @@
...
 import { Icon, CollapseIcon } from "@/components/icon";
 
 import "./collapsible.css";
 
 export interface CollapsibleProps {
     children: ReactNode;
     title: ReactNode;
-    label?: string;
-    isOpen?: boolean;
     className?: string;
```
Comment (vicky-comeau) [inline apps/docs/components/collapsible/Collapsible.tsx:13]: hm not sure why your build error is saying that className is required.

File: apps/docs/components/collapsible/Collapsible.tsx [hunk 2]
```diff
@@ -1,54 +1,31 @@
...
+import { Button, UNSTABLE_Disclosure as Disclosure, UNSTABLE_DisclosurePanel as DisclosurePanel } from "react-aria-components";
 import { Icon, CollapseIcon } from "@/components/icon";
 
 import "./collapsible.css";
 
 export interface CollapsibleProps {
     children: ReactNode;
     title: ReactNode;
-    label?: string;
-    isOpen?: boolean;
```
Comment (vicky-comeau) [inline apps/docs/components/collapsible/Collapsible.tsx:14]: why is this removed? we never want to decide if it's open by default? In that case it would probably be called defaultOpen.  Or the RAC uses defaultExpanded.

File: apps/docs/components/collapsible/Collapsible.tsx [hunk 3]
```diff
@@ -1,54 +1,31 @@
...
 import clsx from "clsx";
-import { type ReactNode, useEffect, useRef, useState } from "react";
-import { ToggleButton } from "react-aria-components";
+import type { ReactNode } from "react";
+import { Button, UNSTABLE_Disclosure as Disclosure, UNSTABLE_DisclosurePanel as DisclosurePanel } from "react-aria-components";
 import { Icon, CollapseIcon } from "@/components/icon";
 
 import "./collapsible.css";
 
 export interface CollapsibleProps {
```
Comment (vicky-comeau) [inline apps/docs/components/collapsible/Collapsible.tsx:10]: Should we not extend the Disclosure properties?

File: apps/docs/components/collapsible/collapsible.css [hunk 4]
```diff
@@ -47,8 +51,6 @@
     transition: height cubic-bezier(0.4, 0, 0.2, 1) 0.4s;
```
[the remark this answers is not in the document]
Comment (fraincs) [reply apps/docs/components/collapsible/collapsible.css:51]: We had them yes. We could wait for disclosure to be out of alpha or do something custom.

File: apps/docs/components/collapsible/Collapsible.tsx (hunk 2 above)
Comment (fraincs) [reply apps/docs/components/collapsible/Collapsible.tsx:14]: I removed them since isOpen was used to open / close the custom disclosure, now that we use RAC component, this is offloaded to them. If you see a use for this I can add it. Since it's for the documentation and there's no current use cases I'm not sure it's worth.

File: apps/docs/components/collapsible/Collapsible.tsx (hunk 3 above)
Comment (vicky-comeau) [reply apps/docs/components/collapsible/Collapsible.tsx:10]: we might as well, then you probably don't need to add most of the props yourself.

Comment (fraincs) [reply apps/docs/components/collapsible/Collapsible.tsx:10]: Exactly

File: apps/docs/components/collapsible/Collapsible.tsx [hunk 5]
```diff
@@ -1,54 +1,31 @@
...
-import { ToggleButton } from "react-aria-components";
-import { Icon, CollapseIcon } from "@/components/icon";
+import type { ReactNode } from "react";
+import { Button, UNSTABLE_Disclosure as Disclosure, UNSTABLE_DisclosurePanel as DisclosurePanel, type DisclosureProps } from "react-aria-components";
 
 import "./collapsible.css";
 
-export interface CollapsibleProps {
+export interface CollapsibleProps extends DisclosureProps {
     children: ReactNode;
```
Comment (vicky-comeau) [inline apps/docs/components/collapsible/Collapsible.tsx:11]: you should be able to remove children and className from here since it's defined inside DisclosureProps and DisclosureProps allows the use of renderProps

Comment (vicky-comeau) [reply apps/docs/components/collapsible/Collapsible.tsx:11]: I see you remove className but not children?

File: apps/docs/components/collapsible/Collapsible.tsx [hunk 6]
```diff
@@ -1,54 +1,37 @@
...
-            contentRef.current.style.height = "0px";
-        }
-    }, [open]);
-
-    const toggle = () => {
-        setOpen(!open);
-    };
+const Collapsible = ({ title, className, children: childrenProp, ...rest }: CollapsibleProps) => {
+    const children = composeRenderProps(childrenProp, prev => {
+        return prev ;
```
Comment (vicky-comeau) [inline apps/docs/components/collapsible/Collapsible.tsx:16]: tiny thing, but there's an extra space

**Review by vicky-comeau — APPROVED** [review]
